gdscript-project.el

```diff
@@ -0,0 +1,111 @@
+      (when (file-exists-p scene-name) scene-name))))
+
+(defun gdscript-project--select-scene ()
+  "Find all scenes files and let user choose one."
+  (let* ((rl (gdscript-util--find-project-configuration-file))
+         (scene-list (mapcar (lambda (x) (file-relative-name x rl)) (directory-files-recursively rl ".*.tscn" t)))
+         (prompt (format "Buffer %s is not scene file, select scene to run" (buffer-name))))
+    (gdscript-util--read scene-list prompt)))
+
+(defun gdscript-project--current-buffer-script ()
+  "Return the name of current script.
+
+If current buffer is not visiting script file return nil."
+  (when buffer-file-name
+    (save-excursion
+      (goto-char (point-min))
+      (when (re-search-forward "^extends SceneTree\\|^extends MainLoop" nil t)
+        (file-relative-name buffer-file-name (gdscript-util--find-project-configuration-file))))))
+
+(defun gdscript-project--select-script ()
+  "Find all script files and let user choose one."
+  (let ((hydra-open gdscript-hydra--open))
+    (when hydra-open (gdscript-hydra--menu/nil))
+    (unwind-protect
+        (let* ((prompt (format "Buffer %s is not script file, select script to run" (buffer-name)))
+               (script-name (gdscript-util--read gdscript-project--script-list prompt)))
+          (gdscript-godot--run-script script-name))
+      (when hydra-open (gdscript-hydra--menu/body)))))
+
+(defun gdscript-project--ag-cleanup ()
+  "Clean after ag search.
+
+Try to leave Emacs as it was before ag search was launched."
+  (remove-hook 'ag-search-finished-hook #'gdscript-project--ag-find-next-script)
+  (let* ((ag-buffer (current-buffer))
+         (ag-window (get-buffer-window ag-buffer))
+         (prev-buffer (window-prev-buffers ag-window)))
+    (if prev-buffer (kill-buffer ag-buffer)
+      (delete-window ag-window)
+      (kill-buffer ag-buffer))))
+
+(defun gdscript-project--ag-find-next-script ()
+  "Find next script file in ag buffer."
+  (let ((pos (next-single-property-change (point) 'compilation-message))
+        (comp-mes (get-text-property (point) 'compilation-message)))
+    (when comp-mes
+      (let ((file-name (caar (compilation--loc->file-struct (compilation--message->loc comp-mes)))))
+        (push file-name gdscript-project--script-list)))
+    (if pos (progn (goto-char pos)
+                   (gdscript-project--ag-find-next-script))
+      (gdscript-project--ag-cleanup)
+      (with-current-buffer (window-buffer (selected-window))
+        (gdscript-project--select-script)))))
+
+(defun gdscript-project--get-all-scripts ()
+  "Find all script files and let user choose one.
+
+Since detection of script files require inspection of file contents,
+this use ag for performance."
+  (if (not (featurep 'ag))
+      (error (format "Buffer %s is no script file. To see all available scripts in current project install package 'ag'." (buffer-name)))
```

Comment:
The error triggers even if the buffer is a gdscript file. Shouldn't it be possible to run files named .gd?